source/client/player/input/Controller.cpp

```diff
@@ -25,29 +27,46 @@ bool Controller::isValidStick(int stickNo)
 
 float Controller::linearTransform(float a1, float a2, float a3, bool b)
 {
-	float x1 = fabsf(a2);
-	float x2 = a1;
-	float x3 = fabsf(x1);
-	if (x3 >= x2)
-		return 0.0f;
+	if (a1 < 0.0)
+		a2 = -a2;
+	if (fabsf(a2) >= fabsf(a1))
+		return 0.0;
+
+	float v5 = (a1 - a2) * a3;
```

Comment:
Lucky for us, that function is never actually called, since the two functions that call it are unused. :) This is fixed in an upcoming commit.
Do you know what this function's purpose is? I'm not too sure.